multipledispatch/variadic.py

```diff
@@ -0,0 +1,96 @@
+import six
+
+from .utils import typename
+
+
+class VariadicSignatureType(type):
+    # checking if subclass is a subclass of self
+    def __subclasscheck__(self, subclass):
+        other_type = (subclass.variadic_type if isvariadic(subclass)
+                      else (subclass,))
+        return subclass is self or all(
+            issubclass(other, self.variadic_type) for other in other_type
+        )
+
+    def __eq__(self, other):
+        """
+        Return True if other has the same variadic type
+
+        Parameters
+        ----------
+        other : object (type)
+            The object (type) to check
+
+        Returns
+        -------
+        bool
+            Whether or not `other` is equal to `self`
+        """
+        return (isvariadic(other) and
+                set(self.variadic_type) == set(other.variadic_type))
+
+    def __hash__(self):
+        return hash((type(self), frozenset(self.variadic_type)))
```

Comment:
No need to call `frozenset` here since `self.variadic_type` is already a `tuple` which is hashable.

Reply:
do we sort this tuple in any way? I believe we want `Variadic[int, float]` and `Variadic[float, int]` to resolve to the same thing.

Reply:
Ah, great point. This is fine then.

Reply:
👍  I used `frozenset` so that `Variadic[(int, float)]` would match `Variadic[(float, int)]`